rfc-123-github-actions-ci.md

```diff
@@ -0,0 +1,102 @@
+- Supported by GitHub, from whom we get Enterprise support via our GitHub Enterprise Cloud subscription.
+- We don't have to host any of our own CI infrastructure.
+- Well documented, easy to [get started][actions-get-started] building a new pipeline. Easier than Concourse for CI as PR builds are built-in to GitHub rather than needing to use [third-party code](https://github.com/telia-oss/github-pr-resource).
+- Doesn't require access to the VPN like our Jenkins does, or membership of any GitHub teams, in order to see logs for public repos.
+- External contributors can see their changes and why they failed, without us having to [push branches][merge-a-pull-request] and paste build logs from Jenkins. This fits more readily with our "code in the open" ethos, and saves on internal developer time.
+- Already in use across some of our production applications. Examples: [business volunteer form][business-volunteer-form], [find support form][find-support-form], [vulnerable people form][vulnerable-people-form]
+- In the future, we can make our internal GitHub Actions into libraries so we’re not duplicating “set up Rails apps” steps across 50 apps and can update them in one place (perhaps `alphagov/actions`?). We already do something similar with `alphagov/govuk-jenkinslib`. See [Homebrew/actions](https://github.com/Homebrew/actions) for an example.
+
+### Cons
+
+- We have less control over what software is on the runners. But most of the time we only need a database, a version of Ruby and some gems. And updating our Jenkins CI instances with new packages is a massive pain.
+- Another system to learn, as well as the replatforming work already ongoing.
+- Duplication of build parameters and steps between all our apps, as they’re all Rails apps. This can be mitigated if we use a shared actions library as proposed earlier.
+
+### Findings from some initial explorations into using GitHub Actions
+
+#### `push` vs `pull_request` build triggers
+
+We considered the following options:
+
+1. Running `on: [pull_request, push]`. This means that internal devs get feedback on their commits when they push them, for example if they've forgotten to run RuboCop locally. And also so that external contributors PRs get CI feedback on their changes. It does mean that for long-running test suites, we could waste compute power, time and the planet's precious natural resources running things twice that don't need to be run twice.
+
+2. Running `on: push`. Actions don't run [by default on forked PRs](https://github.com/alphagov/govuk-docker/pull/337) - it's a security feature that GitHub has implemented. And [when a PR is raised from a fork](https://github.com/alphagov/govuk-docker/pull/337) and the receiving repo only has a push trigger, the tests don't run. So we'd be back to the annoying state that is [our current process for forked PRs][merge-a-pull-request].
```

Comment:
```suggestion
2. Running `on: push`. This would persist a current problem we have with running tests on PRs from external contributors. When [a PR is raised from a fork](https://github.com/alphagov/govuk-docker/pull/337), the tests won't run (needs `pull_request`). Actions also [don't run on forked PRs](https://github.com/alphagov/govuk-docker/pull/337). So we'd be back to the annoying state that is [our current process for forked PRs][merge-a-pull-request].
```